src/System.Net.HttpListener/src/System/Net/Managed/ChunkedInputStream.cs

```diff
@@ -118,11 +118,19 @@ private void OnRead(IAsyncResult base_ares)
             try
             {
                 int nread = base.EndRead(base_ares);
+                if (nread == 0)
+                {
+                    _no_more_data = true;
+                    ares._count = rb.InitialCount - rb.Count;
```

Comment:
I believe _ares.count is intended to hold the total number of bytes read.  If you look at EndRead below, it returns _ares.count.

That said, I agree the logic here looks rather strange...